util/stack/stack.go

```diff
@@ -46,22 +46,23 @@ func Helper() {
 func Traces(err error) []*Stack {
 	var st []*Stack
 
-	wrapped, ok := err.(interface {
-		Unwrap() error
-	})
-	if ok {
-		st = Traces(wrapped.Unwrap())
+	switch e := err.(type) {
+	case interface{ Unwrap() error }:
+		st = Traces(e.Unwrap())
+	case interface{ Unwrap() []error }:
+		for _, ue := range e.Unwrap() {
+			st = Traces(ue)
+			// Only take first stack
```

Comment:
Does this make sense? For nested errors it is expected that there are multiple stacktraces and they are all merged into array here.

Reply:
My worry is that merging the arrays in this case would lead to a stack which doesn't make sense. Essentially this would be the case where you had multiple calls to `errors.Wrap` from either a loop or multiple goroutines running in parallel, then combined using `errors.Join`. The stacks would not be part of a single stack. I think this case would be rare and the case where that is used, you could have multiple of the same stack from parallel processes running into the same error case.

If we wanted to support that, then `[][]*Stack` as the return would make more sense. It doesn't seem like a very relevant edge case to me though.